roles/matrix-base/templates/static-files/well-known/matrix-support.j2

```diff
@@ -0,0 +1,9 @@
+#jinja2: lstrip_blocks: "True"
+{
+	{% if matrix_homeserver_admin_contacts|length > 0 %}
+    "admins": {{ matrix_homeserver_admin_contacts|to_json }}
+	{% endif %}
+	{% if matrix_homeserver_support_url %},
+    "support_page": "{{ matrix_homeserver_support_url }}"
+	{% endif %}
+}
```

Comment:
```suggestion
{
    "admins": {{ matrix_homeserver_admin_contacts|to_json }},
    "support_page": {{ matrix_homeserver_support_url|to_json }}
}
```

This suggestion changes 2 things:

- adds `|to_json` for `matrix_homeserver_support_url` for additional safety
- removes the `if`s, because keeping them makes it hard to reason about the syntax. With the old way if one does not define any contacts (`matrix_homeserver_admin_contacts: []`) but does define a support URL (in `matrix_homeserver_support_url`), the syntax will be off: `{, "support_page": "....."}`.

[The spec](https://github.com/matrix-org/matrix-spec-proposals/pull/1929) does not tell clients what to do if a property exists but is empty. It says `support_page` is optional, but does not distinguish between `support_page` not being defined or being defined as an empty string or `null`.

As far as `admins` is concerned, I think that an empty list there (e.g. `"admins": []`) doesn't hurt, so we can safely drop `{% if matrix_homeserver_admin_contacts|length > 0 %}`.

As a middleground, you may do:

```suggestion
{
    "admins": {{ matrix_homeserver_admin_contacts|to_json }}
    {% if matrix_homeserver_support_url %},
    "support_page": {{ matrix_homeserver_support_url|to_json }}
    {% endif %}
}
```

This way, `admins` will always be there. `support_page` is really optional and may or may not be added. The syntax should be fine with either variable being set.

Reply:
Resolved in [3632c15](https://github.com/spantaleev/matrix-docker-ansible-deploy/pull/1931/commits/3632c158f3f9036fd71bf0ed1b86d286407bb935)